windows build workflow can fail because of rust-cache

It seems that Rust-cache seems to be causing the build workflow to fail (https://github.com/master-of-zen/Av1an/commit/605bad6773025ceae938d48862e9cf99685d1d98) when going from FFmpeg 6.0 to FFmpeg 7.0. This can of course happen every time the ffmpeg version is modified, I don't really know how this can be fixed, maybe by using Rust-cache's prefix-key?
Maybe something like this, so that it doesn't reuse the cache when the FFmpeg version is different.
```
- uses: Swatinem/rust-cache@v2
  with:
    prefix-key: "ffmpeg-${{ env.ffmpeg_ver }}-av1an"
```

For posterity, this is the error when using an old rust-cache with a different ffmpeg version:

<details>
<summary>Error message</summary>

2024-08-18T18:08:11.3927230Z ##[group]Run cargo build --release
2024-08-18T18:08:11.3927626Z [36;1mcargo build --release[0m
2024-08-18T18:08:11.3957715Z shell: C:\Program Files\PowerShell\7\pwsh.EXE -command ". '{0}'"
2024-08-18T18:08:11.3958130Z env:
2024-08-18T18:08:11.3958330Z   CARGO_INCREMENTAL: 0
2024-08-18T18:08:11.3958570Z   ffmpeg_ver: 7.0
2024-08-18T18:08:11.3958813Z   ffmpeg_path: C:/ffmpeg
2024-08-18T18:08:11.3959062Z   vsynth_ver: R69
2024-08-18T18:08:11.3959320Z   vsynth_path: C:/Program Files/Vapoursynth
2024-08-18T18:08:11.3959717Z   FFMPEG_DIR: C:\ffmpeg\ffmpeg-7.0-full_build-shared
2024-08-18T18:08:11.3960072Z   CACHE_ON_FAILURE: false
2024-08-18T18:08:11.3960330Z ##[endgroup]
2024-08-18T18:08:11.7776277Z     Updating crates.io index
2024-08-18T18:08:32.1254782Z    Compiling av1an v0.4.1 (D:\a\Av1an\Av1an\av1an)
2024-08-18T18:08:32.1265886Z    Compiling av1an-core v0.4.1 (D:\a\Av1an\Av1an\av1an-core)
2024-08-18T18:09:37.1372688Z error: linking with `link.exe` failed: exit code: 1181
2024-08-18T18:09:37.1373878Z   |
2024-08-18T18:09:37.1490702Z   = note: "C:\\Program Files\\Microsoft Visual Studio\\2022\\Enterprise\\VC\\Tools\\MSVC\\14.40.33807\\bin\\HostX64\\x64\\link.exe" "/NOLOGO" "C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\rustcUIakCs\\symbols.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.31q9s1gymh37qyq7b62mwhj8e.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.affinity-53bebc2dbcd4b899.affinity.8b854cefdd3d88a5-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.aligned_vec-f2cb9c3acfc59ca7.aligned_vec.d014fd7b625d6892-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.alloc-c032859c81f4576b.alloc.6015141cac529798-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.ansi_term-ab7887104a5be570.ansi_term.51d5c490f8b63999-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.anstream-02d8f0f1ef9d1281.anstream.c5d25686b4ba277e-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.anstyle-6709af08f8e05612.anstyle.b5eabe25e97f807-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.anstyle_parse-fb81a42e5c98b173.anstyle_parse.3ea31dc0be416110-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.anstyle_query-23b83249243373f2.anstyle_query.6f2e33c684135545-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.anstyle_wincon-b5b5ae60ed995e01.anstyle_wincon.47100c2b9c7d8b9e-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.anyhow-4a90d065916f2eca.anyhow.b7f96511fe8fff7e-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.arrayvec-5e8bfee1069e0fab.arrayvec.79c13daf9a7a010f-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.atty-e5cd6a9a234815c9.atty.c20f6612f2fda6e9-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.av1_grain-dfeb1ce1f093ab67.av1_grain.6b79ba5449a76a06-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.av1an.9a0584abcce10c58-cgu.0.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.av1an_core-d50b2554ca9815fb.av1an_core.5005c0f708b730aa-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.av_bitstream-07cd1ead9271ac99.av_bitstream.bf1b21b1abf8a408-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.av_data-d1076a77c0355f6d.av_data.125959b3332b48e1-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.av_format-251f59c3d7e34c6a.av_format.68f1e2ae5a6959da-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.av_ivf-6f538417012db8eb.av_ivf.5723c1cde130c54a-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.av_scenechange-32e4e4cd3f639fd2.av_scenechange.687f8a972819cc1d-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.bitflags-0b72870437d9166f.bitflags.920a6f94393977ef-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.bitflags-167949a2aaedb33c.bitflags.4459fecb987274a3-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.bitstream_io-35f33c7e4dae44b5.bitstream_io.20a5ed9a623d750c-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.byte_slice_cast-e11f2d88e939d5d0.byte_slice_cast.f4ad47c8ab4fcf98-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.byteorder-b737700d3f54346b.byteorder.4ae8b0236acf7dfd-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.bytes-cff8415b06e92e83.bytes.b0eaa0b4fe130e4e-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.cfg_if-8ad2ebec50a855e4.cfg_if.a069446eb7210543-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.cfg_if-c91146a1b584a0a7.cfg_if.d3f2d513fdd9c60e-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.chrono-33a5d3673bd77684.chrono.ee53614e3371e70d-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.clap-1a304da94c13ffd9.clap.a14986932df72cc5-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.clap_builder-ce7b83eb6f9cd805.clap_builder.f7e94d24e7203522-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.clap_lex-97625d2282921168.clap_lex.9dc25740326eb635-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.colorchoice-31b1669b9e4bc81b.colorchoice.77e6a79e7edb4312-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.console-666105e27c415f35.console.f63b3281f0d92ca1-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.core-dfdcb1635a201156.core.911ab0b3a5f9874b-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.crossbeam_channel-bff1eb52345adf4d.crossbeam_channel.261e67f71663eafb-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.crossbeam_deque-44463a5ea591abf3.crossbeam_deque.5638919eb385015e-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.crossbeam_epoch-548e4554bf710b4e.crossbeam_epoch.ccaf1ab23248d1c-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.crossbeam_utils-ae415d0ce6ba7a8b.crossbeam_utils.95814bc4bcac26e1-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.dashmap-ca40a46eef228e71.dashmap.b1d44feb017dc5e7-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.debug_unreachable-2efb81d49a37c6e1.debug_unreachable.844694d81a9ede6b-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.either-57f2e74e195569a7.either.797ffe42c558a852-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.encode_unicode-cf31bbb905ae8e4a.encode_unicode.5fdececcdb81d2f0-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.ffmpeg_sys_the_third-dfb940f6766a6252.ffmpeg_sys_the_third.f118961a62a458a6-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.ffmpeg_the_third-d16442cc38b3cddf.ffmpeg_the_third.1a5aa60a97db8d0f-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.flexi_logger-69c64ded0eef3764.flexi_logger.2a45e7c2efbfdf6e-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.getrandom-0c77d04a22c67bbf.getrandom.38c57985cf3e10c8-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.glob-d24d5d037fc34592.glob.9935c61ec320d7cf-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.hashbrown-34eebc52fd18130a.hashbrown.d4035733e256aa71-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.hashbrown-5e5ab7fb8d3e9a6b.hashbrown.6a483a19f1efea43-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.home-f5a486ebd5bd039b.home.c11299fb793722a9-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.indicatif-6c2134126b5cd12d.indicatif.29e46def24ece3ed-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.is_terminal_polyfill-cc1db7f18936d22d.is_terminal_polyfill.dedec8cd4d120c4a-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.itertools-5db11c1d60d59c8e.itertools.113a20fbab0fca4f-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.itertools-68c1c4603687172a.itertools.84e13bfeaf679931-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.itoa-46c74a6fe538cf6f.itoa.53722b0e1c150a2-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.lazy_static-885333b07ff87f70.lazy_static.ed501b878094284d-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.libc-1014b031c45ff439.libc.fe308fd67a060f70-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.lock_api-042a13c0ffaf4109.lock_api.3b1b92918c7fddc6-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.log-aa8b425b59ec3bce.log.4a8a910fdcc67177-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.maybe_rayon-c4d6fbb06ba756dd.maybe_rayon.20d9a97507dbae52-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.memchr-932c7fb49e3c9ac3.memchr.af2587ae9c367eef-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.mio-7794642eef0360b1.mio.7072cd7c5e6c4c76-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.nom-fb9702b66556546b.nom.d9488b6ec4412e2e-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.ntapi-3db177dfb352acfa.ntapi.5c0547d4fe659e6a-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.nu_ansi_term-7546e220c921b0ca.nu_ansi_term.8a87ce301d3212c3-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.num_bigint-3dcb0fad57efdcd4.num_bigint.82fab54c21e6c470-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.num_cpus-30bba0cfb72de294.num_cpus.ba6f064fa81b8ede-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.num_integer-a28544f93afddb1f.num_integer.46ac6c9c3eccfcc-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.num_rational-54d4d91ce3db88d9.num_rational.c049e74f000827ee-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.num_traits-0ab97a01963a0569.num_traits.1a93b3a504832190-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.number_prefix-359cdaebe573cf30.number_prefix.f0ef79c38c545fb2-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.once_cell-781ed49e495d1d3d.once_cell.5e79b186f39b4fd2-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.panic_unwind-97f6a0482881a03a.panic_unwind.3f950ba578307742-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.parking_lot-86f43b4164b72d68.parking_lot.e1074b8a0f8c8d64-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.parking_lot_core-ef71f39f78f62cb5.parking_lot_core.2c82dacc044b0b33-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.path_abs-144f384f265b63af.path_abs.a648217a46df9528-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.pin_project_lite-bd2bdcec1beff6d5.pin_project_lite.fc7b367535b38d3f-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.plotters-46cc998831e8864b.plotters.c76448d064555c9a-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.plotters_backend-372bccda01181509.plotters_backend.5d0446fe34c05f9d-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.plotters_svg-5b27178f97071385.plotters_svg.ed6d1631fe5082a4-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.portable_atomic-77c69b585f80ebbc.portable_atomic.c1bfcc9c8fcf1104-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.ppv_lite86-80c520d4613efdc1.ppv_lite86.6dc20adcb936821f-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.profiling-ed31c31d69535adc.profiling.61319764dc5f003d-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.rand-d4bf1eb2673e3696.rand.96ed6d2f54ee8b-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.rand_chacha-b8a86194310840f1.rand_chacha.8028076c10af48ab-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.rand_core-f6e04ae9e57a2406.rand_core.946dabc39503f0d5-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.rav1e-0fb6c7621d05aba2.rav1e.ba4513d192b1767e-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.rayon-cc8b6a5ee4a3e7b1.rayon.4b8f23d17fa6e86c-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.rayon_core-25650dd2c3888dc6.rayon_core.96ad350c79e79d85-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.rustc_demangle-f8c4d6a2240f107f.rustc_demangle.2026efdced41830-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.rustc_std_workspace_alloc-7846558dfa99a578.rustc_std_workspace_alloc.66883b82cdc2c970-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.rustc_std_workspace_core-628fee62996a202b.rustc_std_workspace_core.213984d493082cf6-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.ryu-ffe62c756d37e1ee.ryu.66a562635a65fe3e-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.scopeguard-5872f357705fbc39.scopeguard.57f7aea30534a46d-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.serde-8ecff7e422edffb1.serde.33505a241f4b36d5-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.serde_json-02fd0866bb41fca9.serde_json.8e80b8bba1b4a84d-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.shlex-bd72ea56ed5bc744.shlex.888062f4dfcc412f-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.simdutf8-d46ca252353c0f74.simdutf8.f358e57e1e7476f9-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.smallvec-0ed1ef3d93d4bbac.smallvec.1d0d896176f399f8-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.smawk-db4567bb245cd958.smawk.1ceaac47479b53ad-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.socket2-8066667f959a9f3b.socket2.4fedcd3ea6ae8b6c-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.splines-723ee487b5aaa593.splines.2143616f2ca6f691-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.std-d7a86f21fcd377c7.std.f01c98544d5f9dc9-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.std_detect-803b4d5ce4fcd522.std_detect.a49d35c9e5afd7ec-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.std_prelude-811132a6bd31be36.std_prelude.69f14c45bd21aeaf-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.stfu8-a85657926f0d6c14.stfu8.79cdc01e46322fb4-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.strsim-871dd95f53407a05.strsim.a5f778927df64a98-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.strum-376d245eca5382d4.strum.ad1a61ff671ce116-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.sysinfo-2e82f8558f2858a4.sysinfo.3a8748385b69d943-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.textwrap-6a11dfc0bb020ec5.textwrap.377b8a81f75bc1e5-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.thiserror-79fac8a6021b847a.thiserror.8a7300aecef3caa7-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.tokio-07a1940be164195f.tokio.2556c4d0e9fa6ede-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.unicode_linebreak-83226458f36449c2.unicode_linebreak.dc81d696f80fc935-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.unicode_width-fb579f0a88dc6206.unicode_width.ae500dc954d09329-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.unwind-3adc2db30827f7fe.unwind.6b16239f30134d6d-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.utf8parse-59e9782f3c5e9c8a.utf8parse.54901490b02b287d-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.v_frame-61ff64c615f285c9.v_frame.3eec5358c05a39c6-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.vapoursynth-c9a0cc8279fe13d1.vapoursynth.f7071a521d093eb2-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.vapoursynth_sys-5117c51d299169e5.vapoursynth_sys.45c5c2c7c2b499d4-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.which-120d4d2c411b7d08.which.fa13b744f734bb36-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.winapi-fdb9f6e3a33a685a.winapi.d4f040f33c8c4d8a-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.windows-88b620896a4c1308.windows.de7072eb3b75f379-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.windows_core-01b2e6fcf2790fdf.windows_core.54397f4fc90b8bee-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.windows_result-1c42f408e06f0dfb.windows_result.c2e625ac9ba59c60-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.windows_sys-d242c9d1f2e8cdda.windows_sys.c581ca761d984e04-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.windows_targets-e0d73f416f39e7bb.windows_targets.9f8b2249034399bb-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.winsafe-9f01fb46458d9d49.winsafe.103fcfc6aaa80749-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.y4m-6387b5482e4a4f15.y4m.6104abb9b15496c6-cgu.0.rcgu.o.rcgu.o" "D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.zerocopy-aaeb6568892d4224.zerocopy.bddba38e988e4698-cgu.0.rcgu.o.rcgu.o" "/LIBPATH:D:\\a\\Av1an\\Av1an\\target\\release\\deps" "/LIBPATH:C:\\ffmpeg\\ffmpeg-6.0-full_build-shared\\lib" "/LIBPATH:C:\\Program Files\\Microsoft Visual Studio\\2022\\Enterprise\\VC\\Tools\\MSVC\\14.40.33807\\atlmfc\\lib\\x64" "/LIBPATH:D:\\a\\Av1an\\Av1an\\target\\release\\build\\rav1e-b1e5966f77b587fd\\out" "/LIBPATH:C:\\Program Files\\VapourSynth\\sdk\\lib64" "/LIBPATH:C:\\Program Files\\VapourSynth-32\\sdk\\lib64" "/LIBPATH:C:\\Program Files (x86)\\VapourSynth\\sdk\\lib64" "/LIBPATH:C:\\Program Files (x86)\\VapourSynth-32\\sdk\\lib64" "/LIBPATH:C:\\Users\\runneradmin\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\windows_x86_64_msvc-0.52.6\\lib" "/LIBPATH:C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib" "C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\rustcUIakCs\\librav1e-0fb6c7621d05aba2.rlib" "C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\rustcUIakCs\\libstd-d7a86f21fcd377c7.rlib" "C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libcompiler_builtins-1f67c2a5a11a0b2e.rlib" "windows.0.52.0.lib" "ntdll.lib" "windows.0.52.0.lib" "windows.0.52.0.lib" "windows.0.52.0.lib" "advapi32.lib" "kernel32.lib" "ktmw32.lib" "user32.lib" "windows.0.52.0.lib" "bcrypt.lib" "advapi32.lib" "windows.0.52.0.lib" "vapoursynth.lib" "vsscript.lib" "advapi32.lib" "cfgmgr32.lib" "credui.lib" "gdi32.lib" "kernel32.lib" "msimg32.lib" "opengl32.lib" "secur32.lib" "user32.lib" "winspool.lib" "avutil.lib" "avcodec.lib" "avformat.lib" "avdevice.lib" "avfilter.lib" "swscale.lib" "swresample.lib" "legacy_stdio_definitions.lib" "kernel32.lib" "advapi32.lib" "kernel32.lib" "ntdll.lib" "userenv.lib" "ws2_32.lib" "kernel32.lib" "ws2_32.lib" "kernel32.lib" "/defaultlib:msvcrt" "/NXCOMPAT" "/LIBPATH:C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib" "/OUT:D:\\a\\Av1an\\Av1an\\target\\release\\deps\\av1an.exe" "/OPT:REF,ICF" "/DEBUG" "/PDBALTPATH:%_PDB%" "/NATVIS:C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\intrinsic.natvis" "/NATVIS:C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\liballoc.natvis" "/NATVIS:C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\libcore.natvis" "/NATVIS:C:\\Users\\runneradmin\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\libstd.natvis"
2024-08-18T18:09:37.1587165Z   = note: LINK : fatal error LNK1181: cannot open input file 'avutil.lib'
2024-08-18T18:09:37.1587655Z           
2024-08-18T18:09:37.1587782Z 
2024-08-18T18:09:37.1850042Z error: could not compile `av1an` (bin "av1an") due to 1 previous error
2024-08-18T18:09:37.2803084Z ##[error]Process completed with exit code 1.
</details>

The solution you show should work fine. Just FYI, the default prefix key is `rust-v0`, so `rust-${{ env.ffmpeg_ver }}` should be enough to differentiate it from the other caches and still increment with FFmpeg version numbers.

--- a/.github/workflows/windows-build.yml
+++ b/.github/workflows/windows-build.yml
@@ -10,10 +10,9 @@ jobs:
 
     env:
       CARGO_INCREMENTAL: 0
-
-      ffmpeg_ver: "7.0"
+      ffmpeg_ver: "7.0.2"
       ffmpeg_path: "C:/ffmpeg"
-      vsynth_ver: "R69"
+      vsynth_ver: "R70"
       vsynth_path: "C:/Program Files/Vapoursynth"
 
     steps:
@@ -42,7 +41,17 @@ jobs:
           Expand-Archive "$tempFile" "$env:vsynth_path"
 
       - uses: actions/checkout@v4
+
       - uses: Swatinem/rust-cache@v2
+        with:
+          prefix-key: "v${{ env.ffmpeg_ver }}-rust"
+
+      - name: Get latest commit
+        id: get_commit
+        shell: pwsh
+        run: |
+          $commit = git log -1 --pretty=format:"%h: %s (%an)"
+          echo "message=$commit" >> $env:GITHUB_OUTPUT
 
       - name: Av1an build
         run: cargo build --release
@@ -52,6 +61,8 @@ jobs:
         with:
           prerelease: true
           tag_name: latest
+          body: |
+            Commit:
+            ${{ steps.get_commit.outputs.message }}
           files: target/release/av1an.exe
           fail_on_unmatched_files: true
-

--- a/site/src/compiling.md
+++ b/site/src/compiling.md
@@ -6,11 +6,11 @@ You can natively build Av1an on Linux and Windows. Cross-compilation is not supp
 
 To compile Av1an from source, you need the following dependencies:
 
-- [Rust](https://www.rust-lang.org/) (version 1.70.0 or higher)
-- [NASM](https://www.nasm.us/)
-- [clang/LLVM](https://llvm.org/)
-- [FFmpeg](https://ffmpeg.org/)
-- [VapourSynth](https://www.vapoursynth.com/)
+- [Rust](https://www.rust-lang.org) (version 1.70.0 or higher)
+- [NASM](https://www.nasm.us)
+- [clang/LLVM](https://llvm.org)
+- [FFmpeg](https://ffmpeg.org)
+- [VapourSynth](https://www.vapoursynth.com)
 
 On Arch Linux, you can install these dependencies by running
 
@@ -35,21 +35,22 @@ If you just want a current build of Av1an that is newer than the last official r
 
 If you want to build the binary yourself, you will need the following dependencies:
 
-- [Microsoft Visual C++ Build Tools](https://visualstudio.microsoft.com/visual-cpp-build-tools/) - this is a dependency for Rust
+- [Microsoft Visual C++ Build Tools](https://visualstudio.microsoft.com/visual-cpp-build-tools) - this is a dependency for Rust
 - [The Rust toolchain](https://static.rust-lang.org/rustup/dist/x86_64-pc-windows-msvc/rustup-init.exe)
 - [VapourSynth](https://github.com/vapoursynth/vapoursynth/releases/latest) (download the portable version; the installed version could also work)
-- [NASM](https://nasm.us/)
-- [FFmpeg](https://github.com/GyanD/codexffmpeg/releases/download/7.0/ffmpeg-7.0-full_build-shared.7z) (thanks to [gyan](https://github.com/GyanD) for providing these builds)
+- [NASM](https://nasm.us)
+- [FFmpeg](https://github.com/GyanD/codexffmpeg/releases/download/7.0.2/ffmpeg-7.0.2-full_build-shared.7z) (thanks to [gyan](https://github.com/GyanD) for providing these builds)
+- [LLVM](https://releases.llvm.org)
 
 ### FFmpeg setup:
-- Extract the file `ffmpeg-7.0-full_build-shared.7z` to a directory.
-- Create a new environment variable named `FFMPEG_DIR` and set it to the directory path where you extracted `ffmpeg-7.0-full_build-shared.7z`.
-  - (For example, set `FFMPEG_DIR` to `C:\Users\Username\Downloads\ffmpeg-7.0-full_build-shared`)
+- Extract the file `ffmpeg-7.0.2-full_build-shared.7z` to a directory.
+- Create a new environment variable named `FFMPEG_DIR` and set it to the directory path where you extracted `ffmpeg-7.0.2-full_build-shared.7z`.
+  - (For example, set `FFMPEG_DIR` to `C:\Users\Username\Downloads\ffmpeg-7.0.2-full_build-shared`)
 	
 ### VapourSynth setup:
 - Extract the contents of the portable VapourSynth zip file to a directory.
 - Create a new environment variable named `VAPOURSYNTH_LIB_DIR` and set it to the directory path where you extracted the file, appending `\sdk\lib64` to the path.
-  - (For example, set `VAPOURSYNTH_LIB_DIR` to `C:\Users\Username\Downloads\VapourSynth64-Portable-R69\sdk\lib64`)
+  - (For example, set `VAPOURSYNTH_LIB_DIR` to `C:\Users\Username\Downloads\VapourSynth64-Portable-R70\sdk\lib64`)
 
 Then, either clone the repository by running
 
@@ -61,4 +62,4 @@ Or download and extract the [source code](https://github.com/master-of-zen/Av1an
 
 Open a command prompt or PowerShell window inside the cloned repository/extracted ZIP folder and run the command `cargo build --release`. If this command executes successfully with no errors, `av1an.exe` will be in the folder `target\release`.
 
-To use `av1an.exe`, copy all the `.dll` files from `ffmpeg-7.0-full_build-shared\bin` to the same directory as `av1an.exe`, and ensure that `ffmpeg.exe` is in a folder accessible via the `PATH` environment variable.
+To use `av1an.exe`, copy all the `.dll` files from `ffmpeg-7.0.2-full_build-shared\bin` to the same directory as `av1an.exe`, and ensure that `ffmpeg.exe` is in a folder accessible via the `PATH` environment variable.

